WEB-INF/classes/gov/noaa/pfel/erddap/handlers/EDDGridLonPM180Handler.java

```diff
@@ -0,0 +1,122 @@
+package gov.noaa.pfel.erddap.handlers;
+
+import com.cohort.array.StringArray;
+import com.cohort.util.String2;
+import gov.noaa.pfel.erddap.dataset.EDD;
+import gov.noaa.pfel.erddap.dataset.EDDGrid;
+import gov.noaa.pfel.erddap.dataset.EDDGridLonPM180;
+import gov.noaa.pfel.erddap.util.EDStatic;
+import org.xml.sax.Attributes;
+import org.xml.sax.SAXException;
+
+public class EDDGridLonPM180Handler extends State {
+    private StringBuilder content = new StringBuilder();
+    private SaxParsingContext context;
+    private State completeState;
+    private String datasetID;
+
+    public EDDGridLonPM180Handler(SaxHandler saxHandler, String datasetID, State completeState, SaxParsingContext context) {
+        super(saxHandler);
+        this.datasetID = datasetID;
+        this.completeState = completeState;
+        this.context = context;
+    }
+
+    public EDDGrid tChildDataset = null;
+    public String tAccessibleTo = null;
+    public String tGraphsAccessibleTo = null;
+    public boolean tAccessibleViaWMS = true;
+    public boolean tAccessibleViaFiles = EDStatic.defaultAccessibleViaFiles;
+    public StringArray tOnChange = new StringArray();
+    public String tFgdcFile = null;
+    public String tIso19115File = null;
+    public String tDefaultDataQuery = null;
+    public String tDefaultGraphQuery = null;
+    public int tReloadEveryNMinutes = Integer.MAX_VALUE;
+    public int tUpdateEveryNMillis = Integer.MAX_VALUE;
+    public int tnThreads = -1;
+    public boolean tDimensionValuesInMemory = true;
+
+    @Override
+    public void startElement(String uri, String localName, String qName, Attributes attributes) throws SAXException {
+        if (localName.equals("dataset")) {
+            if (tChildDataset == null) {
+                String tType = attributes.getValue("type");
+                if (tType == null || !tType.startsWith("EDDGrid")) {
+                    throw new RuntimeException(
+                            "Datasets.xml error: "
+                                    + "The dataset defined in an "
+                                    + "EDDGridLonPM180 must be a subclass of EDDGrid.");
+                }
+
+                String active = attributes.getValue("active");
+                String childDatasetID = attributes.getValue("datasetID");
+                State state = HandlerFactory.getHandlerFor(tType, childDatasetID, active, this, saxHandler, context);
+                saxHandler.setState(state);
+            } else {
+                throw new RuntimeException(
```

Comment:
I think this else is actually if there's a second child dataset. Should the message be different?

Reply:
Yes that's right, there is a different message for this condition. I missed it, will change it.
Thanks for pointing it out!